.github/workflows/ci-windows.yml

```diff
@@ -10,16 +10,13 @@ jobs:
     outputs:
       dist-key: ${{ steps.cache-keys.outputs.dist }}
     steps:
-    - name: Download LLVM
+    - name: Hash LLVM
       run: |
-        Invoke-WebRequest -Uri "https://github.com/llvm/llvm-project/releases/download/llvmorg-17.0.2/llvm-project-17.0.2.src.tar.xz" -OutFile llvm.tar.xz
-        $hash = (Get-FileHash llvm.tar.xz).Hash
-        unxz llvm.tar.xz
-        tar -x --strip-components=1 -f llvm.tar
-        echo "hash=$hash" >> $env:GITHUB_OUTPUT
+        $url = "https://github.com/llvm/llvm-project/releases/download/llvmorg-17.0.5/llvm-project-17.0.5.src.tar.xz"
+        echo "llvmurl=$url" >> $env:GITHUB_OUTPUT
       id: download
     - name: Generate cache keys
-      run: echo "dist=${{ runner.os }}-llvm-${{ steps.download.outputs.hash }}" >> $env:GITHUB_OUTPUT
+      run: echo "dist=${{ runner.os }}-llvm-${{ steps.download.outputs.llvmurl }}" >> $env:GITHUB_OUTPUT
```

Comment:
Better to hash the URL instead of using the URL itself so the key will not be too long.

Reply:
Not very pretty doing a hash through powershell xS